flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/KafkaSourceBuilder.java

```diff
@@ -497,6 +520,29 @@ private void sanityCheck() {
                         "Property %s is required when offset commit is enabled",
                         ConsumerConfig.GROUP_ID_CONFIG));
         // Check offsets initializers
+        if (startingOffsetsInitializer == null) {
+            if (props.get(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG) != null) {
+                startingOffsetsInitializer =
+                        OffsetsInitializer.committedOffsets(
```

Comment:
I believe, the function `sanityCheck()` is for checking the validity of the settings, but not to change them. Shouldn't we instead update the settings in `parseAndSetRequiredProperties()`?

Reply:
Modified, thank you.